[Mobile Payments] Fix blurry payment method icons in payment methods screen

jaclync opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

Closes WOOMOB-140

- [x] ⚠️ Please review https://github.com/Automattic/Gridicons-iOS/pull/82 first.
- [x] ⚠️ @jaclync updates Podfile to point Gridicons pod version to merge commit in trunk after above PR is merged

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This pull request fixes the blurry icons in the payment methods view. When using `UIImage` in SwiftUI, the image does not automatically scale with dynamic type. In order to fix the blurry images using Gridicons, which currently only supports `UIImage` and the image assets aren't accessible externally, I opened a PR in Gridicons to add SwiftUI Image support https://github.com/Automattic/Gridicons-iOS/pull/82.

### Dependency Updates:
* [`Podfile`](diffhunk://#diff-8f7d6adf31268a2d897ee34bd170592648d6e520aa237104395e4a4438af50cbL76-R77): Updated the `Gridicons` dependency to point to the latest PR commit https://github.com/Automattic/Gridicons-iOS/pull/82. Once the Gridicons PR is approved and merged, I will update the Podfile again.

### UI Improvements:
* [`WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift`](diffhunk://#diff-80eacec4b691bde1f01beb6d9d0cd0b2eb9892ddac50f2987e9cf7c86dd022ccL55-R75): Replaced Gridicons `UIImage` references for payment method icons (e.g., `.priceImage`, `.creditCardImage`, `.linkImage`) with `Image.gridicon` calls. This enables auto-scaling of the Gridicons.

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

Prerequisite: the store has an order pending payment, eligible for card payments, and has a payment link

- Go to the orders tab
- Open the order in the prerequisite
- Tap `Collect Payment` --> in any font size, the icon for "Cash", "Card Reader", and "Share Payment Link" rows should not look blurry

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

I tested all payment methods rows by enabling all rows in code.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

before | after
-- | --
![Simulator Screenshot - iPad (10th generation) - 2025-05-21 at 15 05 40](https://github.com/user-attachments/assets/98074aa6-3aa4-4a13-baab-f8cae71e5ead) | ![Simulator Screenshot - iPad (10th generation) - 2025-05-21 at 15 01 27](https://github.com/user-attachments/assets/fd207d2d-7db4-40c9-b7e2-ee584b6ee608)



---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Point Gridicons pod to PR commit.

jaclync committed:
Use Gridicon SwiftUI Image version for icons in payment methods view in SwiftUI so that icons do not lose resolution from UIImage to SwiftUI conversion.

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

jaclync committed:
Update release notes.

staskus reviewed (approved):
All works well 👍 

I think only `GridIcon` changes need to also be reflected in a template file so `rake gen` wouldn't overwrite them.

jaclync committed:
Merge branch 'trunk' into feat/WOOMOB-140-fix-blurry-payment-method-icons

# Conflicts:
#	Podfile.lock

jaclync committed:
Point Gridicons pod to the latest PR commit.

jaclync commented:
Updated the Gridicons pod commit to the latest PR commit in https://github.com/woocommerce/woocommerce-ios/pull/15659/commits/bc181dd6adbe7ff077e4fee438e532901e440d12 and fixed merge conflicts from Podfile.lock from trunk. Tested it again and the icons remain not blurry:

<img src="https://github.com/user-attachments/assets/24ef05f1-5017-4f0a-bc2c-ff4e35b07f42" width="500" />

staskus reviewed (approved):
Everything looks good now! Thanks.

jaclync committed:
Point Gridicons pod to PR merge commit.

jaclync committed:
Merge branch 'trunk' into feat/WOOMOB-140-fix-blurry-payment-method-icons

# Conflicts:
#	Podfile.lock
